drivers/SmartThings/philips-hue/src/init.lua

```diff
@@ -830,38 +831,137 @@ local function do_bridge_network_init(driver, device, bridge_url, api_key)
+                        new_device_info = hue_device
+                        break
+                      end
+                    end
+                    if new_device_info ~= nil then break end
+                  end
+
+                  if new_device_info == nil then
+                    log.warn("Couldn't get all device info for new light, unable to join. Try using Scan Nearby to find new Hue lights.")
+                    return
+                  end
+
+                  log.info(
+                    string.format(
+                      "Adding light \"%s\" to Philips Hue Edge Driver",
+                      add_data.metadata.name
+                    )
+                  )
+
+                  local profile_ref
+
+                  if add_data.color then
+                    if add_data.color_temperature then
+                      profile_ref = "white-and-color-ambiance"
+                    else
+                      profile_ref = "legacy-color"
+                    end
+                  elseif add_data.color_temperature then
+                    profile_ref = "white-ambiance" -- all color temp products support `white` (dimming)
+                  elseif add_data.dimming then
+                    profile_ref = "white"          -- `white` refers to dimmable and includes filament bulbs
+                  else
+                    log.warn(
+                      string.format(
+                        "Light resource [%s] does not seem to be A White/White-Ambiance/White-Color-Ambiance device, currently unsupported"
+                        ,
+                        add_data.id
+                      )
+                    )
+                    return
+                  end
+
+                  local create_device_msg = {
+                    type = "EDGE_CHILD",
+                    label = add_data.metadata.name,
+                    vendor_provided_label = new_device_info.product_data.product_name,
+                    profile = profile_ref,
+                    manufacturer = new_device_info.product_data.manufacturer_name,
+                    model = new_device_info.product_data.model_id,
+                    parent_device_id = bridge_device.id,
+                    parent_assigned_child_key = add_data.id,
+                  }
+
+                  Discovery.light_state_disco_cache[add_data.id] = {
+                    hue_provided_name = add_data.metadata.name,
+                    id = add_data.id,
+                    on = add_data.on,
+                    color = add_data.color,
+                    dimming = add_data.dimming,
+                    color_temperature = add_data.color_temperature,
+                    mode = add_data.mode,
+                    parent_device_id = bridge_device.id,
+                    hue_device_id = add_data.owner.rid,
+                    hue_device_data = new_device_info
+                  }
+
+                  driver:try_create_device(create_device_msg)
```

Comment:
Mostly curious: What happens to the device if the add fails? Like this event is an sse that a new device has been added are we just back at the original behavior?

Reply:
If this fails, there's no way for the driver to know. So it would either get picked up at the next driver restart or the next "Scan Nearby".